[10] iExpenses/iExpenses/Views/AddExpenseView/AddExpenseViewModel.swift

```diff
@@ -0,0 +1,46 @@
+//
+//  AddExpenseViewModel.swift
+//  iExpenses
+//
+//  Created by Joe Pham on 2022-08-04.
+//
+
+import Foundation
+
+/// Handles hosting data during the creation of a new expense,
+/// used with a `@StateObject` property wrapper.
+///
+/// The "heavy lifting" operations (i.e. actually adding an `ExpenseItem` to an array)
+/// is done by `ExpenseListViewModel` instead, using `@ObservedObject`
+final class AddExpenseViewModel: ObservableObject {
+	
+	private let listViewModel: ExpenseListViewModel
+	
+	init(parentVM: ExpenseListViewModel) {
+		self.listViewModel = parentVM
+	}
```

Comment:
I like this single responsibility refactor, compared to previous case the add expense view has to have knowledge of 2 different view models